### ISSUE TITLE
The finetuning hyperparameters of resnet50

### ISSUE THREAD

**Issue by Vickeyhw** [keyu-tian/SparK#27]
The hyperparameter settings (batch size and learning rate) in the paper seem inconsistent with the code. Which could reproduce the performance (80.6 acc) reported in the paper?

**Comment by Vickeyhw** [comment]
@keyu-tian I have tried this setting except that batch size is 2048, for the reason that 8 gpus cannot accommodate 4096 images. Unfortunately, my training losses became NaN. Does the batch size affect so much?

**Comment by keyu-tian** [comment]
@Vickeyhw can u provide the command and logs?

**Comment by keyu-tian** [comment]
@Vickeyhw can u provide the command and logs?

**Comment by Vickeyhw** [comment]
The args are:
```
(nstream_imagenet/main.py, line  29)=> initial args:
{'base_lr': 0.002,
 'batch_size_per_gpu': 256,
 'best_val_acc': 0.0,
 'bs': 2048,
 'clip': -1,
 'cmd': '--local_rank=0 --exp_name res50 --exp_dir '
        './output//Spark_res50_official_finetune_default_lr2e-3_ld0_7_300e '
        '--data_path=imagenet2012/ImageNet_ILSVRC2012 '
        '--model=resnet50 --ep=300 --bs=2048 '
        '--resume_from=./resnet50_1kpretrained_timm_style.pth',

 'cur_ep': '',
 'data_path': '/home/bingxing2/public/imagenet2012/ImageNet_ILSVRC2012',
 'dataloader_workers': 8,
 'device': device(type='cuda', index=0),
 'dist_on_itp': False,
 'dist_url': 'env://',
 'drop_path': 0.05,
 'ema': 0.9999,
 'ep': 300,
 'eval_data_path': '',
 'exp_dir': 'output/Spark_res50_official_finetune_default_lr2e-3_ld0_7_300e',
 'exp_name': 'res50',
 'finish_time': '',
 'first_logging': True,
 'glb_batch_size': 2048,
 'global_rank': 0,
 'img_size': 224,
 'is_local_master': True,
 'is_master': True,
 'local_rank': 0,
 'log_epoch': <bound method FineTuneArgs.log_epoch of FineTuneArgs(prog='main.py', usage=None, description=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)>,
 'log_txt_name': 'output/Spark_res50_official_finetune_default_lr2e-3_ld0_7_300e/finetune_log.txt',
 'lr': 0.016,
 'lr_scale': 0.7,
 'mixup': 0.1,
 'model': 'resnet50',
 'opt': 'lamb',
 'remain_time': '',
 'rep_aug': 0,
 'resume_from': './resnet50_1kpretrained_timm_style.pth',
 'sbn': False,
 'tb_lg_dir': '/home/bingxing2/home/scx6008/hw/SparK/output/Spark_res50_official_finetune_default_lr2e-3_ld0_7_300e/tensorboard_log',
 'train_acc': 0.0,
 'train_loss': 0.0,
 'wd': 0.02,
 'world_size': 8,
 'wp_ep': 5}
``` 
The logs are:
```
{"name": "res50", "cmd": "--local_rank=0 --exp_name res50  --exp_dir ./output//Spark_res50_official_finetune_default_lr2e-3_ld0_7_300e --data_path=imagenet2012/ImageNet_ILSVRC2012 --model=resnet50 --ep=300 --bs=2048 --resume_from=./resnet50_1kpretrained_timm_style.pth",  "model": "resnet50"}

{"cur_ep": "", "train_L": 0.0, "train_acc": 0.0, "best_val_acc": 0.0, "rema": "", "fini": ""}
{"cur_ep": "1/300", "train_L": 0.15892334485948087, "train_acc": 0.62875, "best_val_acc": 4.741999879479408, "rema": "1 day, 3:49:28", "fini": "04-18 14:07"}
{"cur_ep": "1/300", "train_L": 0.1589116028137505, "train_acc": 0.67875, "best_val_acc": 4.741999879479408, "rema": "1 day, 3:49:55", "fini": "04-18 14:07"}
{"cur_ep": "2/300", "train_L": 0.006166074915975333, "train_acc": 17.9, "best_val_acc": 4.741999879479408, "rema": "23:23:38", "fini": "04-18 09:46"}
{"cur_ep": "2/300", "train_L": 0.006192157221585512, "train_acc": 17.850625, "best_val_acc": 4.741999879479408, "rema": "23:23:11", "fini": "04-18 09:45"}
{"cur_ep": "3/300", "train_L": 0.005531815142929554, "train_acc": 31.03125, "best_val_acc": 4.741999879479408, "rema": "23:10:48", "fini": "04-18 09:37"}
{"cur_ep": "3/300", "train_L": 0.005463680490851402, "train_acc": 31.53375, "best_val_acc": 4.741999879479408, "rema": "23:10:48", "fini": "04-18 09:37"}
{"cur_ep": "4/300", "train_L": 0.005408975677192211, "train_acc": 34.206875, "best_val_acc": 4.741999879479408, "rema": "23:16:55", "fini": "04-18 09:48"}
{"cur_ep": "4/300", "train_L": 0.0053789736032485965, "train_acc": 34.44125, "best_val_acc": 4.741999879479408, "rema": "23:16:55", "fini": "04-18 09:48"}
{"cur_ep": "5/300", "train_L": 0.005379272519052029, "train_acc": 34.675, "best_val_acc": 4.741999879479408, "rema": "23:14:28", "fini": "04-18 09:50"}
{"cur_ep": "5/300", "train_L": 0.005380104035139084, "train_acc": 34.819375, "best_val_acc": 4.741999879479408, "rema": "23:14:28", "fini": "04-18 09:50"}
{"cur_ep": "6/300", "train_L": 0.005402451483905315, "train_acc": 34.683125, "best_val_acc": 43.54200065135956, "rema": "1 day, 0:28:58", "fini": "04-18 11:10"}
{"cur_ep": "6/300", "train_L": 0.00541508878916502, "train_acc": 34.975625, "best_val_acc": 43.54200065135956, "rema": "1 day, 0:29:01", "fini": "04-18 11:10"}
{"cur_ep": "7/300", "train_L": 0.005425414913892746, "train_acc": 35.530625, "best_val_acc": 43.54200065135956, "rema": "22:59:30", "fini": "04-18 09:45"}
{"cur_ep": "7/300", "train_L": 0.005408091994374991, "train_acc": 35.624375, "best_val_acc": 43.54200065135956, "rema": "22:59:30", "fini": "04-18 09:45"}
{"cur_ep": "8/300", "train_L": 0.005414580475538969, "train_acc": 35.54125, "best_val_acc": 43.54200065135956, "rema": "22:52:39", "fini": "04-18 09:43"}
{"cur_ep": "8/300", "train_L": 0.005382590828090906, "train_acc": 35.548125, "best_val_acc": 43.54200065135956, "rema": "22:52:39", "fini": "04-18 09:43"}
{"cur_ep": "9/300", "train_L": 0.005471015560626983, "train_acc": 34.43, "best_val_acc": 43.54200065135956, "rema": "22:54:35", "fini": "04-18 09:50"}
{"cur_ep": "9/300", "train_L": 0.005475558865070343, "train_acc": 34.46375, "best_val_acc": 43.54200065135956, "rema": "22:54:35", "fini": "04-18 09:50"}
{"cur_ep": "10/300", "train_L": 0.005533626601845026, "train_acc": 33.153125, "best_val_acc": 43.54200065135956, "rema": "22:50:06", "fini": "04-18 09:50"}
{"cur_ep": "11/300", "train_L": 0.005674835596978664, "train_acc": 30.861875, "best_val_acc": 43.54200065135956, "rema": "1 day, 0:08:31", "fini": "04-18 11:13"}
{"cur_ep": "11/300", "train_L": 0.00564453030526638, "train_acc": 30.945, "best_val_acc": 43.54200065135956, "rema": "1 day, 0:08:34", "fini": "04-18 11:13"}
{"cur_ep": "12/300", "train_L": 0.005810007998347282, "train_acc": 28.496875, "best_val_acc": 43.54200065135956, "rema": "22:46:05", "fini": "04-18 09:56"}
{"cur_ep": "12/300", "train_L": 0.005785862766951323, "train_acc": 28.413125, "best_val_acc": 43.54200065135956, "rema": "22:46:05", "fini": "04-18 09:56"}
{"cur_ep": "13/300", "train_L": 0.005920035427808761, "train_acc": 25.194375, "best_val_acc": 43.54200065135956, "rema": "22:33:55", "fini": "04-18 09:48"}
{"cur_ep": "13/300", "train_L": 0.005946401672065258, "train_acc": 25.08125, "best_val_acc": 43.54200065135956, "rema": "22:33:55", "fini": "04-18 09:48"}
{"cur_ep": "14/300", "train_L": 0.006212903738021851, "train_acc": 19.32875, "best_val_acc": 43.54200065135956, "rema": "22:36:47", "fini": "04-18 09:56"}
{"cur_ep": "14/300", "train_L": 0.006215040449798107, "train_acc": 19.2525, "best_val_acc": 43.54200065135956, "rema": "22:36:47", "fini": "04-18 09:56"}
{"cur_ep": "15/300", "train_L": 0.006856135655939579, "train_acc": 8.3275, "best_val_acc": 43.54200065135956, "rema": "22:27:26", "fini": "04-18 09:51"}
}
{"cur_ep": "16/300", "train_L": 0.00847176744043827, "train_acc": 0.505, "best_val_acc": 43.54200065135956, "rema": "23:35:44", "fini": "04-18 11:04"}
{"cur_ep": "16/300", "train_L": 0.00849286153614521, "train_acc": 0.4875, "best_val_acc": 43.54200065135956, "rema": "23:36:30", "fini": "04-18 11:05"}
{"cur_ep": "17/300", "train_L": 0.4059472487047315, "train_acc": 0.17875, "best_val_acc": 43.54200065135956, "rema": "21:58:07", "fini": "04-18 09:31"}
{"cur_ep": "17/300", "train_L": 0.4075041047215462, "train_acc": 0.1925, "best_val_acc": 43.54200065135956, "rema": "21:58:55", "fini": "04-18 09:32"}
{"cur_ep": "18/300", "train_L": 2.594160234707594, "train_acc": 0.20125, "best_val_acc": 43.54200065135956, "rema": "21:50:27", "fini": "04-18 09:28"}
{"cur_ep": "18/300", "train_L": 2.599315336358547, "train_acc": 0.19, "best_val_acc": 43.54200065135956, "rema": "21:50:27", "fini": "04-18 09:28"}
{"cur_ep": "19/300", "train_L": 168.27923201017379, "train_acc": 0.213125, "best_val_acc": 43.54200065135956, "rema": "21:47:55", "fini": "04-18 09:30"}
{"cur_ep": "19/300", "train_L": 168.41776486291886, "train_acc": 0.20375, "best_val_acc": 43.54200065135956, "rema": "21:47:55", "fini": "04-18 09:30"}
{"cur_ep": "20/300", "train_L": 9278.141375096131, "train_acc": 0.196875, "best_val_acc": 43.54200065135956, "rema": "21:40:22", "fini": "04-18 09:28"}
{"cur_ep": "20/300", "train_L": 9242.067996807862, "train_acc": 0.225625, "best_val_acc": 43.54200065135956, "rema": "21:40:25", "fini": "04-18 09:28"}
{"cur_ep": "21/300", "train_L": 385708.6738011719, "train_acc": 0.193125, "best_val_acc": 43.54200065135956, "rema": "22:55:53", "fini": "04-18 10:48"}
{"cur_ep": "21/300", "train_L": 384956.72077226563, "train_acc": 0.216875, "best_val_acc": 43.54200065135956, "rema": "22:55:53", "fini": "04-18 10:48"}
{"cur_ep": "22/300", "train_L": 4882009.3119, "train_acc": 0.20625, "best_val_acc": 43.54200065135956, "rema": "21:27:03", "fini": "04-18 09:24"}
{"cur_ep": "22/300", "train_L": 4842194.24495, "train_acc": 0.22, "best_val_acc": 43.54200065135956, "rema": "21:27:06", "fini": "04-18 09:24"}
{"cur_ep": "23/300", "train_L": 51736270.4836, "train_acc": 0.209375, "best_val_acc": 43.54200065135956, "rema": "21:28:39", "fini": "04-18 09:30"}
{"cur_ep": "23/300", "train_L": 52044852.8624, "train_acc": 0.169375, "best_val_acc": 43.54200065135956, "rema": "21:28:42", "fini": "04-18 09:30"}
{"cur_ep": "24/300", "train_L": 122264843.6944, "train_acc": 0.22125, "best_val_acc": 43.54200065135956, "rema": "21:21:00", "fini": "04-18 09:27"}
{"cur_ep": "25/300", "train_L": 462680769.6992, "train_acc": 0.184375, "best_val_acc": 43.54200065135956, "rema": "21:16:28", "fini": "04-18 09:27"}
{"cur_ep": "25/300", "train_L": 464397992.0096, "train_acc": 0.211875, "best_val_acc": 43.54200065135956, "rema": "21:16:30", "fini": "04-18 09:27"}
{"cur_ep": "26/300", "train_L": 1761032724.8896, "train_acc": 0.198125, "best_val_acc": 43.54200065135956, "rema": "22:27:24", "fini": "04-18 10:43"}
{"cur_ep": "26/300", "train_L": 1763697504.5888, "train_acc": 0.188125, "best_val_acc": 43.54200065135956, "rema": "22:27:51", "fini": "04-18 10:43"}
{"cur_ep": "27/300", "train_L": NaN, "train_acc": 0.20375, "best_val_acc": 43.54200065135956, "rema": "20:57:51", "fini": "04-18 09:18"}
{"cur_ep": "27/300", "train_L": NaN, "train_acc": 0.211875, "best_val_acc": 43.54200065135956, "rema": "20:58:21", "fini": "04-18 09:18"}
{"cur_ep": "28/300", "train_L": NaN, "train_acc": 0.19125, "best_val_acc": 43.54200065135956, "rema": "20:49:07", "fini": "04-18 09:14"}
```

**Comment by keyu-tian** [comment]
i see, i would check that. Perhaps I copied the wrong code of lamb optimizer.

BTW, have you tried a ConvNeXt-small? Would it fail too?

**Comment by Vickeyhw** [comment]
@keyu-tian  ConvNeXt-small seems normal so far.

**Comment by Vickeyhw** [comment]
@keyu-tian Have you found the resnet-50 fine-tuning problem?  The ConvNeXt-small reaches 83.96 validation acc fine-tuning from your released pertaining weights.
